src-tauri/src/command/mod.rs

```diff
@@ -28,31 +28,46 @@ pub struct LinkScoreMap {
 /// Create a `.link`
 async fn create_link(
     url: String,
-    state: tauri::State<'_, Cli>,
+    meta_data: arklib::link::Metadata
 ) -> Result<String> {
     let url = Url::parse(url.as_str())?;
     let resource = arklib::id::ResourceId::compute_bytes(url.as_ref().as_bytes())
         .map_err(|_| CommandError::Arklib)?;
     let domain = url.domain().expect("Url has no domain");
-    let path = format!("{}/{domain}-{}.link", &state.path, resource.crc32);
+    let file_name = format!("{domain}-{resource}.link");
+    let path = PathBuf::from(ARK_SHELF_WORKING_DIR.get().unwrap()).join(&file_name);
     // Validate there is not already a resource identical
     if std::fs::metadata(&path).is_ok() {
         Err(CommandError::LinkExist)
     } else {
         std::fs::write(path, url.as_str())?;
-        Ok(format!("{domain}-{}.link", resource.crc32))
+        let path = METADATA_PATH.get().unwrap().join(format!("{resource}"));
+        std::fs::write(&path, serde_json::to_string(&meta_data).unwrap())?;
+        Ok(file_name)
     }
 }
 
 #[tauri::command]
 /// Remove a `.link` from directory
-async fn delete_link(name: String, state: tauri::State<'_, Cli>) -> Result<()> {
-    fs::remove_file(format!("{}/{}", &state.path, name))?;
+async fn delete_link(name: String) -> Result<()> {
+    let path = ARK_SHELF_WORKING_DIR.get().unwrap().join(&name);
+    fs::remove_file(&path)?;
+    let resource = name.rsplit("-").enumerate().take_while(|(u,_)| {
```

Comment:
This looks fragile.. We should construct `id` only from the content, i.e. from the link itself (URL / file content).